src/batched/KokkosBatched_Gemm_Decl.hpp

```diff
@@ -76,155 +119,242 @@ struct Gemm {
-                                  ResultsPerThread::Rank2>::type;
-
-    // Selects whether to use the register blocking or non-register blocking
-    // implementation of SerialGemm.
-    using mode_type = typename std::conditional<
-        is_vector,
-        typename std::conditional<on_gpu || on_intel, Algo::Gemm::Blocked,
-                                  Algo::Gemm::Unblocked>::type,
-        typename std::conditional<
-            on_gpu, Algo::Gemm::Unblocked,
-            typename std::conditional<on_a64fx, Algo::Gemm::Unblocked,
-                                      Algo::Gemm::Blocked>::type>::type>::type;
-
-#if 0
-      std::cout << "view_scalar_type:" << typeid(view_scalar_type).name() << std::endl <<
-  	        "execution_space:" << typeid(typename CViewType::execution_space).name() << std::endl <<
-                "resultsPerThread:" << typeid(resultsPerThread).name() << std::endl <<
-                "mode_type:" << typeid(mode_type).name() << std::endl <<
-                "is_vector:" << is_vector << std::endl <<
-                "on_gpu:" << on_gpu << std::endl <<
-                "on_intel:" << on_intel << std::endl <<
-                "on_a64fx:" << on_a64fx << std::endl;
-#endif
-    ret = BatchedSerialGemm<ArgTransA, ArgTransB, mode_type, ArgBatchLayout,
-                            resultsPerThread>::invoke(alpha, A, B, beta, C);
-    return ret;
+  // Select whether to calculate a rank-0 or rank-2 result per thread
+  using resultsPerThread =
+      typename std::conditional<!is_vector && on_gpu, ResultsPerThread::Rank0,
+                                ResultsPerThread::Rank2>::type;
+
+  // Selects whether to use the register blocking or non-register blocking
+  // implementation of SerialGemm.
+  using mode_type = typename std::conditional<
+      is_vector,
+      typename std::conditional<on_gpu || on_x86_64, Algo::Gemm::Blocked,
+                                Algo::Gemm::Unblocked>::type,
+      typename std::conditional<
+          on_gpu, Algo::Gemm::Unblocked,
+          typename std::conditional<on_a64fx, Algo::Gemm::Unblocked,
+                                    Algo::Gemm::Blocked>::type>::type>::type;
+
+  if (handle->enableDebug) {
+    std::cout << "view_scalar_type:" << typeid(view_scalar_type).name()
+              << std::endl
+              << "execution_space:"
+              << typeid(typename CViewType::execution_space).name() << std::endl
+              << "resultsPerThread:" << typeid(resultsPerThread).name()
+              << std::endl
+              << "mode_type:" << typeid(mode_type).name() << std::endl
+              << "is_vector:" << is_vector << std::endl
+              << "on_gpu:" << on_gpu << std::endl
+              << "on_x86_64:" << on_x86_64 << std::endl
+              << "on_a64fx:" << on_a64fx << std::endl;
   }
-};
+
+  switch (handle->get_kernel_algo_type()) {
+    case BaseKokkosBatchedAlgos::KK_SERIAL:
```

Comment:
Just to make sure I understand it correctly: the interface currently only supports ```KK_SERIAL```, doesn't it?

Reply:
Correct.